src/json-patch-duplex.ts

```diff
@@ -145,66 +154,43 @@ module jsonpatch {
   /* The operations applicable to object root. Many are the same as for the object */
   var rootOps = {
     add: function (obj) {
-      rootOps.remove.call(this, obj);
+      var removed = rootOps.remove.call(this, obj);
       for (var key in this.value) {
         if (this.value.hasOwnProperty(key)) {
           obj[key] = this.value[key];
         }
       }
-      return true;
+      return removed;
```

Comment:
I think in this case we should return entre object that was removed.

If I get code right, this will return array of all properties of this root object.